### ISSUE TITLE
Failed to unmarshal manifest: error unmarshaling JSON: json: cannot unmarshal string into Go value of type map[string]interface {}

### ISSUE THREAD

**Issue by srehmanproov** [argoproj/argo-cd#3294]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a
question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [ ] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

Creating a application with valid yaml 

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

**Expected behavior**

**Screenshots**

Unable to create application: application spec is invalid: InvalidSpecError: Unable to generate manifests in helm/Dev: rpc error: code = Unknown desc = Failed to unmarshal manifest: error unmarshaling JSON: json: cannot unmarshal string into Go value of type map[string]interface {}

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by jannfis** [comment]
Can you please be a little more precise?

**Comment by alexmt** [comment]
Based on our slack conversation one of the manifests in the app git repo is incorrect. Created an enhancement request to simplify finding the file: https://github.com/argoproj/argo-cd/issues/3295

Lets continue troubleshooting it in slack please. Github issue is not very effective way of doing it.

**Comment by Interaze** [comment]
To anyone running into this issue. I found it best summarized in slack by @crenshaw-dev in slack. Here's the message:
Michael Crenshaw
 [1 year ago](https://cloud-native.slack.com/archives/C01TSERG0KZ/p1646925819261439?thread_ts=1646925346.688039&cid=C01TSERG0KZ)

 the CMP server pulls everything from standard out. So if the commands previous to helm template are writing to standard out, they could be causing problems. I’d recommend moving those commands to the “init” command section.

I had an issue of authenticating to a private helm repo. With this I needed to move my helm dependency build and it works.
